janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/AdjacentVertexOptimizerStrategy.java

```diff
@@ -56,21 +51,18 @@
     }
 
     protected void optimizeStep(T step) {
-        P<?> predicate = parsePredicate(step);
-        if (isValidPredicate(predicate)) {
-            OptimizablePosition pos = getOptimizablePosition(step);
-            replaceSequence(step, pos, predicate);
+        OptimizablePosition pos = getOptimizablePosition(step);
+        if (pos != OptimizablePosition.NONE && isValidStep(step)) {
+            replaceSequence(step, pos);
```

Comment:
Can you profile these optimizations if possible? They could take some time (especially the index selection part). Profiling also makes it clear to users whether this optimization is applied or not.

Reply:
By profiling, do you mean to investigate the output of `query.profile()`? Or something else that I miss?

Reply:
Oh I mean you should make the query.profile() result contain the optimization latency here.